plugins/pipelines/app/models/concerns/samson_pipelines/stage_concern.rb

```diff
@@ -1,13 +1,27 @@
 module SamsonPipelines::StageConcern
-  # Return true if any stages in the pipeline are marked production
   def production?
-    super || next_stages.any?(&:production?)
+    super || all_next_stages.any? { |next_stage| next_stage.production? && !next_stage.no_code_deployed? }
   end
 
   def next_stages
     Stage.find(next_stage_ids)
   end
 
```

Comment:
making next_stages private blows up stuff in EnvironmentVariableGroupsController